app/src/helpers/getContractAddresses.ts

```diff
@@ -0,0 +1,42 @@
+import { Contract } from '@prisma/client';
+import { parse, visit } from '@solidity-parser/parser'
+import {AddressInfo} from '~/types'
+
+const isAddress = (val: string) => {
```

Comment:
You can use the [`viem` function](https://viem.sh/docs/utilities/isAddress.html) here as well